syntax = "proto3";
package ecommerce;
import "shared.proto";

service Buyer {
    rpc SearchProduct(SearchProductRequest) returns (BuyerProductResponse);
    rpc BuyProduct(BuyProductRequest) returns (RateProductResponse);
    rpc RateProduct(RateProductRequest) returns (RateProductResponse);
    rpc AddToWishlist(AddToWishlistRequest) returns (RateProductResponse);
}

message BuyProductRequest {
  string product_id = 1;
  int32 qty = 2;
}

message RateProductRequest {
    string product_id = 1;
    int32 rating = 2;
}

message SearchProductRequest {
    string product_name = 1;
    Category category = 2;
}

message AddToWishlistRequest {
    string product_id = 1; 
}

message BuyerProductDetails {
  string product_id = 1;
  string desc = 2;
  int32 qty = 3;
  double rating = 4;
  string product_name = 5;
  Category category = 6;
  double price = 7;
  string seller_addr = 8;
}

message BuyerProductResponse {
  repeated BuyerProductDetails products = 1;
}

message RateProductResponse {
    string status = 1;
}